tests/std/tests/P0009R18_mdspan_extents/test.cpp

```diff
@@ -0,0 +1,294 @@
+constexpr void do_check_members(index_sequence<Indices...>) {
+    using Ext = extents<IndexType, Extents...>;
+
+    // Each specialization of extents models regular and is trivially copyable
+    static_assert(regular<Ext>);
+    static_assert(is_trivially_copyable_v<Ext>);
+
+    // Check member types
+    static_assert(same_as<typename Ext::index_type, IndexType>);
+    static_assert(same_as<typename Ext::size_type, make_unsigned_t<IndexType>>);
+    static_assert(same_as<typename Ext::rank_type, size_t>);
+
+    // Check static observers
+    static_assert(Ext::rank() == sizeof...(Extents));
+    static_assert(Ext::rank_dynamic() == ((Extents == dynamic_extent) + ...));
+    static_assert(((Ext::static_extent(Indices) == Extents) && ...));
+
+    // Check noexceptness of static observers
+    static_assert(noexcept(Ext::rank()));
+    static_assert(noexcept(Ext::rank_dynamic()));
+    static_assert(noexcept(Ext::static_extent(0)));
+
+    // Check default constructor
+    Ext ext;
+    static_assert(is_nothrow_default_constructible_v<Ext>);
+
+    // Check 'extent' observer
+    assert((((ext.extent(Indices) == Extents && Extents != dynamic_extent) || ext.extent(Indices) == 0) && ...));
+
+    using OtherIndexType = conditional_t<is_signed_v<IndexType>, long long, unsigned long long>;
+
+    { // Check construction from other extents
+        using Ext2 = extents<OtherIndexType, Extents...>;
+        Ext2 ext2{ext};
+        assert(((ext.extent(Indices) == ext2.extent(Indices)) && ...));
+        assert(ext == ext2);
+        static_assert(is_nothrow_constructible_v<Ext2, Ext>);
+        // Other tests are defined in 'check_construction_from_other_extents' function
+    }
+
+    { // Check construction from extents pack
+        using Ext2 = extents<OtherIndexType, Extents...>;
+        Ext2 ext2{ext.extent(Indices)...};
+        assert(((ext.extent(Indices) == ext2.extent(Indices)) && ...));
+        assert(ext == ext2);
+        static_assert(is_nothrow_constructible_v<Ext2, Ext>);
+        // Other tests are defined in 'check_construction_from_extents_pack' function
+    }
+
+    { // Check construction from array and span
+        using Ext2 = extents<OtherIndexType, Extents...>;
+
+        auto arr = to_array<IndexType>({ext.extent(Indices)...});
+        Ext ext2a{arr};
+        assert(((ext.extent(Indices) == ext2a.extent(Indices)) && ...));
+        assert(ext == ext2a);
+        static_assert(is_nothrow_constructible_v<Ext2, decltype(arr)>);
+
+        span s{arr};
+        Ext ext2b{s};
+        assert(((ext.extent(Indices) == ext2b.extent(Indices)) && ...));
+        assert(ext == ext2b);
+        static_assert(is_nothrow_constructible_v<Ext2, decltype(s)>);
+        // Other tests are defined in 'check_construction_from_array_and_span' function
+    }
+}
+
+template <class IndexType, size_t... Extents>
+constexpr void check_members() {
+    do_check_members<IndexType, Extents...>(make_index_sequence<sizeof...(Extents)>{});

```

Comment:
This could use `index_sequence_for`.

Reply:
`index_sequence_for<Extents...>` would not work, since it expects types (not `size_t`s)

Reply:
Ah, duh! Thanks, never mind then :joy_cat: